From a3d74a3e861e003836dccfb135f45e924c92b5ad Mon Sep 17 00:00:00 2001
From: Michael Lutz <michi@icosahedron.de>
Date: Wed, 19 Dec 2012 23:31:59 +0100
Subject: -Fix [FS#5398]: [NewGRF] Incorrect values are better than a crash
 when a NewGRF queries vehicle variable 4C before vehicle
 initialisation is completed.

---
 src/train_cmd.cpp |    2 --
 1 files changed, 0 insertions(+), 2 deletions(-)

diff --git a/src/train_cmd.cpp b/src/train_cmd.cpp
index c7bad3b..366d7cd 100644
--- a/src/train_cmd.cpp
+++ b/src/train_cmd.cpp
@@ -373,8 +373,6 @@ int Train::GetCurrentMaxSpeed() const
 	if (_settings_game.vehicle.train_acceleration_model == AM_ORIGINAL) return min(this->gcache.cached_max_track_speed, this->current_order.max_speed);
 
 	int max_speed = this->tcache.cached_max_curve_speed;
-	assert(max_speed == this->GetCurveSpeedLimit());
-
 	if (IsRailStationTile(this->tile)) {
 		StationID sid = GetStationIndex(this->tile);
 		if (this->current_order.ShouldStopAtStation(this, sid)) {
-- 
1.7.9

